### ISSUE TITLE
It should be possible to run inspections for selected modules

### ISSUE THREAD

**Issue by MDoerner** [rubberduck-vba/Rubberduck#3043]
Currently, after every parse we run all inspections although on reparse, only inspections for modules that got reparse or reresolved cam really change. That is rather inefficient, which is made worse by the fact that inspections currently do not get canceled. 

We should add a possibility to run inspections only for selected modules (in case of project wide inspections, for the projects of selected modules) and cache the inspection results. That should reduce the runtime on reparse considerably.

**Comment by retailcoder** [comment]
Definitely! I think I'm going to merge my incomplete stuff in... quite important refactorings and modifications to inspections have been waiting in my fork forever (I need a `TimeFactory`) and I'd like new contributors that would like to tackle  and PR an easy inspection to get acquainted with RD's mechanics.
